ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js

```diff
@@ -108,28 +118,45 @@ class BaseLine extends React.Component {
       return false
     }
   }
+  validatePoint(point) {
+    if (
+      point.length !== 2 ||
+      Number.isNaN(Number.parseFloat(point[0])) ||
+      Number.isNaN(Number.parseFloat(point[1])) ||
+      point[0] > 180 ||
+      point[0] < -180 ||
+      point[1] > 90 ||
+      point[1] < -90
+    ) {
+      return JSON.stringify(point) + ' is not a valid point.'
+    }
+    return ''
+  }
   validateListOfPoints(coordinates) {
     let message = ''
-    if (this.props.mode === 'poly' && coordinates.length < 4) {
-      message = 'Minimum of 4 points needed for polygon'
-    } else if (this.props.mode === 'line' && coordinates.length < 2) {
-      message = 'Minimum of 2 points needed for line'
+    const isLine = this.props.mode.includes('line')
+    const numPoints = isLine ? 2 : 4
+    if (
+      !this.props.mode.includes('multi') &&
+      !coordinates.some(coords => coords.length > 2) &&
+      coordinates.length < numPoints
+    ) {
+      message = `Minimum of ${numPoints} points needed for ${
+        isLine ? 'Line' : 'Polygon'
+      }`
     }
-    coordinates.forEach(point => {
-      if (
-        point.length !== 2 ||
-        (Number.isNaN(Number.parseFloat(point[0])) &&
-          Number.isNaN(Number.parseFloat(point[1])))
-      ) {
-        message = JSON.stringify(point) + ' is not a valid point.'
+    coordinates.forEach(coordinate => {
+      if (coordinate.length > 2) {
+        coordinate.forEach(coord => {
+          if (this.validatePoint(coord)) {
+            message = this.validatePoint(coord)
+          }
+        })
       } else {
-        if (
-          point[0] > 180 ||
-          point[0] < -180 ||
-          point[1] > 90 ||
-          point[1] < -90
-        ) {
-          message = JSON.stringify(point) + ' is not a valid point.'
+        if (this.props.mode.includes('multi')) {
+          message = `Switch to ${isLine ? 'Line' : 'Polygon'}`
```

Comment:
✏️ I'm not sure what this message is for but I find the ternary a bit confusing. If it's currently a line (`isLine` is true) then I'd think the message would be `Switch to Polygon`.

Reply:
Oh I can see how this is confusing with the naming of the variable. This is here for when/if there's capability to choose multi modes, like `MultiPolygon` or `MultiLine` and you have a two-shape wkt where one of the shapes is invalid (has too few points to be a complete shape). The invalid shape is filtered out in lines 148-154 and the overall string is corrected to a valid individual shape. That's why it informs you to switch to the correct single-shape mode, so that the shape and the mode match. This is definitely a corner case that won't happen often, but needed to be addressed so that the user isn't shown an unrelated default error message